preprocessors/togeojson-kml.preprocessor.js

```diff
@@ -102,17 +102,34 @@ module.exports = function(infile, outdirectory, callback) {
       return callback(invalid('KML does not contain any valid features'));
     }
 
-    // Create metadata file for original gpx source
+    // Create metadata file for original kml source
     var metadatafile = path.join(outdirectory, '/metadata.json');
     digest(infile, function(err, metadata) {
       fs.writeFile(metadatafile, JSON.stringify(metadata), function(err) {
-        if (err) return callback(err);
-        return createIndices(callback);
+        if (err) throw err;
+        createIndices(function(err) {
+          if (err) throw err;
+          archiveOriginal(function(err) {
+            if (err) throw err;
+            return callback(err);
```

Comment:
Same thing here - if `err` is defined, it will already have been thrown, so this will likely return an undefined variable.